assessment.md

```diff
@@ -40,43 +40,130 @@ Replace TODOs with abbreviations, or emoji, or ...
 
 * Calculating code coverage of a `pytest` test suite
 
-  TODO: add description for each assessment level for this skill.
+ *None = Does not attempt to implement any test features with pytest.
+
+ *Inadequate = Low code coverage due to inefficient coding.
+
+ *Adequate = Having average code coverage and making attempt to improve code coverage by efficient coding.
+
+ *Good = Having high code coverage on a fairly regular basis and knowing how to further improve code coverage.
+
+ *Excellent = Always having code coverage, ensuring maximum software efficiency.
 
 * Reporting code coverage through an online provider
 
-  TODO: add description for each assessment level for this skill.
+ *None = Not knowing what code coverage is or how to use online provider to report code coverage.
+
+ *Inadequate = Knowing how to report code coverage but not doing it very often
+
+ *Adequate = Knowing how to report code coverage and report at a fairly regular basis.
+
+ *Good =  Reporting code coverage at a regular basis and making attempts to improve code coverage through efficient coding.
+
+ *Excellent = Reporting code coverage at a regular basis and constantly improving software with efficient coding.
 
 * Using docstrings to document a Python program
 
-  TODO: add description for each assessment level for this skill.
+ *None = Does not attempt to use docstrings for documentation
+
+ *Inadequate = Uses docstrings but pays no attention to detail and does not properly document the program within them.
+
+ *Adequate = Uses docstrings that contain basic documentation that allow for a better understanding of the program.
+
+ *Good = Uses docstrings that contain a high-level overview of the program and allow for a good understanding of the program flow.
+
+ *Excellent = Uses docstrings that allow for readers to gain a high-level understanding of program flow and allows for further work to be done on code without any previous knowledge of the project itself.
 
 * Adopting and applying appropriate naming conventions
 
-  TODO: add description for each assessment level for this skill.
+ *None = Uses unclear and not concise naming conventions to convey what the
+programmer is doing through the whole program
+
+ *Inadequate = Uses clear and concise naming conventions to convey what the
+programmer is doing for a small portion of the program
+
+ *Adequate = Uses clear and concise naming conventions to convey what the
+programmer is doing for some of the program
+
+ *Good = Uses clear and concise naming conventions to convey what the
+programmer is doing for most of the program
+
+ *Excellent  Utilizes clear and concise naming conventions to convey
+what the programmer is doing throughout the whole program
```

Comment:
I think it is important to point out that students are expected to apply and adhere to a naming convention, for example, the use of the snake case in python.